<launch>
  <!-- All Arugements Definition -->
    <!-- Args of Tf Frames -->
    <!-- No need to change -->
    <arg name="base_frame_"   value="base" />
    <arg name="laser_frame_" value="laser" />
    <arg name="odom_frame_" value="odom" />
    <arg name="pose_topic_pub_" value="pose" />
    <arg name="laser_topic_pub_" value="laser" />
    <arg name="path_topic_pub_" value="path" />

    <!-- Topics Arguments -->
    <!-- Change to the topic names in the rosbag or others -->
    <arg name="laser_topic_"  value="/robot/front_laser/scan" />
    <arg name="wheel_odom_topic_"  value="/robot/throttle/odom" />
    <!-- Other Predefined parameters -->
    <arg name="use_wheel_odom_" value="true" />
    <arg name="use_laser_" value="true" />
    <arg name="use_wheel_odom_prior_guess_" value="true" />
    <arg name="use_darko_cfg_" value="true" />

  <!-- Some static tf configuration -->
  <!-- Example: static_transform_publisher x y z yaw pitch roll frame_id child_frame_id period_in_ms -->
  <group if="$(arg use_darko_cfg_)">
    <node pkg="tf" type="static_transform_publisher" name="robot_to_sensor" args="0.421 -0.327 0.044 -0.785 0 3.142 $(arg base_frame_) $(arg laser_frame_) 10"/>
  </group>
  <group unless="$(arg use_darko_cfg_)">
    <node pkg="tf" type="static_transform_publisher" name="robot_to_sensor" args="0 0 0 0 0 0 $(arg base_frame_) $(arg laser_frame_) 10"/>
  </group>

  <!-- The node of slam  -->
  <node pkg="slam_demo" name="odometer" type="odom_node" output="screen">
    <param name="use_darko_cfg_"   value="$(arg use_darko_cfg_)" />
    <param name="use_wheel_odom_prior_guess_" value="$(arg use_wheel_odom_prior_guess_)" />
    <param name="use_wheel_odom_" value="$(arg use_wheel_odom_)" />
    <param name="use_laser_" value="$(arg use_laser_)" />
    <param name="base_frame_"   value="$(arg base_frame_)" />
    <param name="laser_frame_" value="$(arg laser_frame_)" />
    <param name="odom_frame_" value="$(arg odom_frame_)" />
    <param name="laser_topic_" value="$(arg laser_topic_)"/>
    <param name="wheel_odom_topic_" value="$(arg wheel_odom_topic_)"/>
  </node>

  <!-- <node pkg="rosbag" type="play" name="rosbag" -->
        <!-- args="-d 5 /home/ros/catkin_ws/src/darko/slam2d/rosbag/b0-2014-07-11-10-58-16.bag" /> -->

   <node pkg="rviz" type="rviz" name="rviz" args="-d $(find slam_demo)/rviz/odom.rviz"/>
</launch>
